### PR TITLE
Echo Reduction

### PR DESCRIPTION
**Pull request by wchmzn**
**Issue #:** 

**Description of changes:**
Add Amazon Chime Echo Reduction for React SDK

**Testing**
1. Have you successfully run `npm run build:release` locally? yes

2. How did you test these changes?
Testing steps:
- Enable Web Audio. Enable Echo Reduction. Join the meeting
- Select one input audio device, and then click on "Enable voice focus". 
- Expect: ns_es model get download and exactly once.
- demo PR: https://github.com/aws-samples/amazon-chime-sdk/pull/128
Echo Reduction On:
Screen Shot 2021-12-01 at 9 23 56 PM
Echo Reduction Off:
Screen Shot 2021-12-01 at 9 24 13 PM
Default Voice Focus On:
Screen Shot 2021-12-06 at 9 51 23 PM

3. If you made changes to the component library, have you provided corresponding documentation changes?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jing-chen1 — COMMENTED** [review]
Thanks for all your work! I like the detailed description and clean code.

You may also want to add your use case and update the description in the storybook docs, such as `AudioInputVFControl.stories.mdx`, `VoiceFocusProvider.stories.mdx` and `useVoiceFocus.stories.mdx`.  Since the feature has additional charges, better to have it documented in the Readme of React Component Library.

File: src/providers/VoiceFocusProvider/index.tsx [hunk 1]
```diff
@@ -49,7 +55,7 @@ const VoiceFocusProvider: React.FC<Props> = ({ spec, options, children }) => {
       prev: VoiceFocusSpec | undefined,
       next: VoiceFocusSpec | undefined
     ): boolean => {
-      if (Object.is(prev, next)) {
+      if (Object.is(prev, next) || JSON.stringify(prev) === JSON.stringify(next)) {
```
[the remark this answers is not in the document]
Comment (wchmzn) [reply src/providers/VoiceFocusProvider/index.tsx:58]: Yeah, the previous compare is shallow one which just compares the pointers. We need to deep compare here to see if the object is really the same.

File: src/providers/VoiceFocusProvider/index.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ interface Props {
    * For more info, you can go to https://aws.github.io/amazon-chime-sdk-js/interfaces/voicefocusdeviceoptions.html
    */
   options?: VoiceFocusDeviceOptions;
+  createMeetingResponse?: any;
```
[the remark this answers is not in the document]
Comment (wchmzn) [reply src/providers/VoiceFocusProvider/index.tsx:24]: sure

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+### Added
+- Allow builders to supply the response from a `CreateMeeting` or `CreateMeetingWithAttendees` call when adding a `VoiceFocusProvider` to the component tree. This enables optional features like Amazon Chime Echo Reduction to be added to devices when turning on Amazon Chime Voice Focus.
```
[the remark this answers is not in the document]
Comment (wchmzn) [reply CHANGELOG.md:12]: nice catch

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ All notable changes to this project will be documented in this file.
...
+
```
[the remark this answers is not in the document]
Comment (wchmzn) [reply CHANGELOG.md:13]: added

**Review by jing-chen1 — DISMISSED** [review]

File: src/types/index.ts [hunk 5]
```diff
@@ -74,3 +74,16 @@ export enum DeviceLabels {
...
+}
+
+export type CreateMeetingResponse = {
+  MeetingFeatures: MeetingFeatures;
+}
+
+export type JoinMeetingInfo = {
+  Meeting: CreateMeetingResponse;
+  Attendee: string;
+}
```
[the remark this answers is not in the document]
Comment (wchmzn) [reply src/types/index.ts]: added

**Review by jing-chen1 — APPROVED** [review]

**Review by jing-chen1 — APPROVED** [review]

**Pull request merged**